controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -563,3 +572,51 @@ func firstNodeNotMatchingName(name string, nodes []corev1.Node) *corev1.Node {
 	}
 	return nil
 }
+
+// modifyImageTag takes an image string, and returns an updated tagged image
+func modifyImageTag(image, tagName string) (string, error) {
```

Comment:
Can we move this (and similar generic functions) to the util package so we can make sure others can use it as well?